### PR TITLE
fix privacy collection status label being incorrect

### LINKED ISSUE #12972: When editing a child of a private collection - the privacy status label is incorrect
### Issue Summary

When editing a collection that's a child collection of one that is private, the privacy is inherited from that parent.

However, the privacy status shows that it is public. When clicking this is does advise the user that it's actually private due to the parent collection. The label and the dialog are not in sync, they show conflicting information.

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. As an admin user, log in to the Wagtail admin
3. Navigate via Settings > Collections & click 'Add a collection'
4. Make a new collection (e.g. 'my private collection'), the parent does not matter for this one
5. Then, edit that newly created collection, change the privacy to any non-public privacy setting (e.g. user access only), click save
6. Now create a new collection, selecting that previously created one as the parent, (e.g. 'is this private?')
7. Now edit that second created collection
8. Expected: The child collection of a private parent collection should show that the status is set to 'Private' (similar to what we do for pages), when clicking this status, the dialog states that the collection is private due to the parent setting.
9. Actual: The status shows 'Public', when clicking this status, the dialog states that the collection is private due to the parent setting.

This causes a confusing situation where the status label says 'Public' when in fact it is both 'Private' and not actually editable directly.

We should reflect the actual status of this sub-collection and show it as 'Private'.

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (no)

Image

Image

### Working on this

- See a previous similar issue for pages - https://github.com/wagtail/wagtail/issues/12277 & the fix https://github.com/wagtail/wagtail/pull/12622
- Anyone can contribute to this that is confident to fix an issue with privacy & Django templates.
- View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

### PR DESCRIPTION
**Pull request by smark-1**
fixes #12972

### CODE REVIEW

**Comment by smark-1** [comment]
@lb- This fix updates the test_collection_is_public tag. But doing a search it seems like the tag is only used in the `templates/wagtailadmin/collection/_privact_switch.html` so it can be refactored more to delete the tag altogether and replace the template like this. 

```
{% load i18n wagtailadmin_tags %}

{% if not collection.is_root %}
    {% url 'wagtailadmin_collections:set_privacy' collection.id as privacy_url %}
    {% dialog icon_name='view' id='set-privacy' data_url=privacy_url title=_("Change privacy") message_status='info' message_heading=_("Privacy settings determine who is able to view documents in this collection.") %}
    {% enddialog %}

    <div
        class="privacy-indicator {% if collection.get_view_restrictions %}private{% else %}public{% endif %}"
        data-controller="w-zone"
        data-action="w-privacy:changed@document->w-zone#switch"
        data-w-zone-active-class="public"
        data-w-zone-inactive-class="private"
        data-w-zone-switch-key-value="isPublic"
    >
        {% trans "Privacy" %}
        <button type="button" data-a11y-dialog-show="set-privacy" class="button button-small button-secondary">
            {# labels are shown/hidden in CSS according to the 'private' / 'public' class on view-permission-indicator #}
            <span class="label-public" aria-label="{% trans 'Set collection privacy. Current status: Public' %}">
                {% icon name="view" classname="default" %}
                {% trans "Public" %}
            </span>
            <span class="label-private" aria-label="{% trans 'Set collection privacy. Current status: Private' %}">
                {% icon name="no-view" classname="default" %}
                {% trans "Private" %}
            </span>
        </button>
    </div>
{% endif %}
```

I am assuming that the template tag was created before the get_view_restriction method existed or maybe it is there to do caching? If it is for the caching wouldn't it be better to cache the get_view_restrictions method?

Not sure which approach is better. Let me know your thoughts.

**Comment by lb-** [comment]
Let's keep the template tag for now, that way we have less change. It will be helpful if we want to show the privacy status in listings in the future and could help add any edge case logic that makes sense in the tag only.

**Comment by lb-** [comment]
I'm sure you're on it but please also add a test for this fix, one that tests against the edit collection view when private and maybe another for when public.

**Review by lb- — APPROVED** [review]
Thank you @smark-1  - this is working great for me locally. I will force push a rebase and check the CI is happy and then merge.

**Comment by lb-** [comment]
OK, error is part of this PR. I will see if it's a quick fix, otherwise @smark-1  you will need to delete your local branch and fix up.

```
ERROR: setUpClass (wagtail.tests.test_collection_model.TestCollectionViewPrivacy)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.12.9/x64/lib/python3.12/site-packages/django/test/testcases.py", line 1400, in setUpClass
    cls.setUpTestData()
  File "/home/runner/work/wagtail/wagtail/wagtail/tests/test_collection_model.py", line 139, in setUpTestData
    get_user_model().objects.create_superuser(
  File "/home/runner/work/wagtail/wagtail/wagtail/test/emailuser/models.py", line 32, in create_superuser
    return self._create_user(email, password, True, True, **extra_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/wagtail/wagtail/wagtail/test/emailuser/models.py", line 17, in _create_user
    user = self.model(
           ^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.9/x64/lib/python3.12/site-packages/django/db/models/base.py", line 567, in __init__
    raise TypeError(
TypeError: EmailUser() got unexpected keyword arguments: 'username'
```

File: wagtail/tests/test_collection_model.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
+from django.contrib.auth import get_user_model
 from django.test import TestCase
+from django.urls import reverse
 
 from wagtail.models import Collection
+from wagtail.models.media import CollectionViewRestriction
```
Comment (lb-) [inline wagtail/tests/test_collection_model.py:8]: OK, we should be putting these tests in the view tests. see `wagtail/admin/tests/test_collections_views.py`.

When you look at other Wagtail view tests, we have a lot of helper classes to avoid the boilerplate of creating admin users etc. 

I have just moved the tests over and here's what it could look like.

```py
class TestCollectionPrivacyStatusLabels(WagtailTestUtils, TestCase):

    def setUp(self):
        self.user = self.login()

        self.root_collection = Collection.get_first_root_node()
        self.public_collection = self.root_collection.add_child(name="Public_photos1")
        self.private_collection = self.root_collection.add_child(name="Private_photos1")
        CollectionViewRestriction.objects.create(
            collection=self.private_collection,
            restriction_type=CollectionViewRestriction.LOGIN,
        )
        self.root_collection.refresh_from_db()
        self.private_collection.refresh_from_db()
        self.public_collection.refresh_from_db()

        self.sub_private_collection = self.private_collection.add_child(
            name="Sub_private_photos1"
        )

        self.sub_public_collection = self.public_collection.add_child(
            name="Sub_public_photos1"
        )

    def test_admin_displays_private_tag_for_private_base_collections(self):
        request = self.client.get(
            reverse("wagtailadmin_collections:edit", args=[self.private_collection.pk])
        )

        self.assertTrue('class="privacy-indicator private"' in str(request.content))
        self.assertFalse('class="privacy-indicator "' in str(request.content))

    def test_admin_displays_private_tag_for_private_sub_collections(self):
        request = self.client.get(
            reverse(
                "wagtailadmin_collections:edit", args=[self.sub_private_collection.pk]
            )
        )

        self.assertTrue('class="privacy-indicator private"' in str(request.content))
        self.assertFalse('class="privacy-indicator public"' in str(request.content))

    def test_admin_displays_public_tag_for_public_base_collections(self):
        request = self.client.get(
            reverse("wagtailadmin_collections:edit", args=[self.public_collection.pk])
        )

        self.assertTrue('class="privacy-indicator public"' in str(request.content))
        self.assertFalse('class="privacy-indicator private"' in str(request.content))

    def test_admin_displays_public_tag_for_public_sub_collections(self):
        request = self.client.get(
            reverse(
                "wagtailadmin_collections:edit", args=[self.sub_public_collection.pk]
            )
        )

        self.assertTrue('class="privacy-indicator public"' in str(request.content))
        self.assertFalse('class="privacy-indicator private"' in str(request.content))

```

**Review by lb- — COMMENTED** [review]
OK, easy fix, the tests just need to a different location and leverage the other base test classes we have. This abstracts some nuances across different DB structures for user creation.
